### PR TITLE
 Depoly helm charts on eks

### PR DESCRIPTION
**Pull request by ehabfeki**
- [x] add helm module ~> deploys Grafana & Prometheus
- [x] add eks module
- [x] add terraform-deploy [jobs: validate-terraform & deploy-terraform]
- [x] test helm charts see [PR-09](https://github.com/ehabshaaban/awesome-eyes/pull/9)

### CODE REVIEW

**Comment by omarahm3** [comment]
احا حسن جامد نيك

File: .terraform.lock.hcl [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# This file is maintained automatically by "terraform init".
```
[the remark this answers is not in the document]
Comment (ehabshaaban) [reply .terraform.lock.hcl:1]: Lock file should be versioned.

When terraform init is working on installing all of the providers needed for a configuration, Terraform considers both the version constraints in the configuration and the version selections recorded in the lock file. Please see [Dependency Lock File](https://www.terraform.io/docs/language/dependency-lock.html)

File: test/infra_test.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	// https://github.com/kubernetes/client-go/blob/master/examples/out-of-cluster-client-configuration/main.go#L64
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), v1.ListOptions{FieldSelector: "status.phase=Running"})
+	if err != nil {
+		log.Fatalf("Error getting EKS pods: %v", err)
+	}
+	log.Printf("There are %d pods associated with cluster: %s", len(pods.Items), *result.Cluster.Name)
+
+	return pods
+}
+
```
[the remark this answers is not in the document]
Comment (ehabshaaban) [reply test/infra_test.go:130]: Thank you for this. I believe this will remove some work in `infra_test.go`.
Will implement it in next iteration, resolving for now.

File: test/infra_test.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+func verifyPrometheus(statusCode int, body string) bool {
+	if statusCode != 200 {
+		return false
+	}
+
+	return strings.Contains(body, "Prometheus Time Series Collection and Processing Server")
+}
+
+func ValidatePrometheusDeployment(wg *sync.WaitGroup, t *testing.T, clientset *kubernetes.Clientset, result *eks.DescribeClusterOutput) {
```
[the remark this answers is not in the document]
Comment (ehabshaaban) [reply test/infra_test.go:158]: Yes!
I think also we need to change it not to be public, I'm not accessing `ValidatePrometheusDeployment` from another file.

File: test/infra_test.go [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+const namespace string = "awesomeeyes"
+
+var kubeOptions *k8s.KubectlOptions = k8s.NewKubectlOptions("", "", namespace)
+
+// Setup a TLS configuration to submit with the helper, a blank struct is acceptable
+var tlsConfig tls.Config = tls.Config{}
+
+// Generate clientset to authenticate k8s cluster
+// https://stackoverflow.com/questions/60547409/unable-to-obtain-kubeconfig-of-an-aws-eks-cluster-in-go-code/60573982#60573982
+func newClientset(cluster *eks.Cluster) (*kubernetes.Clientset, error) {
```
[the remark this answers is not in the document]
Comment (ehabshaaban) [reply test/infra_test.go:44]: `aws_eks_cluster` doesn't generate config file. I can connect to the cluster with `aws eks update-kubeconfig --region region-code --name cluster-name` But that won't be possible when running the test file on CI.

How you can auth the cluster right after [L241](https://github.com/ehabshaaban/awesome-eyes/blob/c2e29feb6b24bfbb31556bf06e9f6aac9708e091/test/infra_test.go#L241) ?

File: modules/eks/eks-cluster.tf [hunk 5]
```diff
@@ -0,0 +1,45 @@
+locals {
+  cluster_name = "awesome-eyes-eks-${random_string.suffix.result}"
```
[the remark this answers is not in the document]
Comment (ehabshaaban) [reply modules/eks/eks-cluster.tf:2]: I'm not quite sure I understand over here. Can you rephrase please?

File: test/infra_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ehabshaaban) [reply test/infra_test.go:158]: Will implement it in next iteration, resolving for now.

File: modules/eks/eks-cluster.tf (hunk 5 above)
[the remark this answers is not in the document]
Comment (mohammedx3) [reply modules/eks/eks-cluster.tf:2]: `cluster_name = "awesome-eyes-eks-${var.clusterName}"`
then create random string and pass it as `clusterName `in the test.
